backend/static/frontend/org/riceapps/controllers/schedule_planner_controller.js

```diff
@@ -83,6 +84,27 @@ SchedulePlannerController.prototype.onCRNViewClick_ = function(event) {
 };
 
 /**
+ * Event handler; called when clear playground button is clicked. Clears all courses from the playground.
+ * @param {SchedulePlannerEvent} event
+ * @private
+ */
+SchedulePlannerController.prototype.onClearPlaygroundClick_ = function(event) {
+  if (this.userModel_ != null && event.type == SchedulePlannerEvent.Type.CLEAR_PLAYGROUND_CLICK){
+    var playground_view = this.view_.getPlaygroundView();
```

Comment:
Hmm... I'm not sure exactly about the modifying the list of children by removing children while iterating.

I think it would be best to just do the calls to removeCoursesFromPlayground() inside of the loop, and then you can call playground_view.removeChildren(true) after the loop to get rid of all of the children at once.